### PR TITLE
document: sort options

### LINKED ISSUE #262: Document Search
- [x] The end user should be able to sort by:
  - [x] number of loans
  - [x] number of available items

- [x] The end user should be able to filter (facet) by:
  * has at least a physical copy
  * has an e-book
  * has no copies
- [x] Available items
  * not sure what available means 
  * in the list of results, I see a col. "Items" but I am not sure it corresponds to the sorting. The col. label should match with the sorting value
- [ ] Most loaned
   - [x] Display number of loans
   - [ ] Respect default values for sort order `asc`, `desc`

https://codimd.web.cern.ch/0p-ZhZDnTAycgWGNULBXBQ#UC-User-can-sort-record-search-results-based-on-circulation-information
https://codimd.web.cern.ch/0p-ZhZDnTAycgWGNULBXBQ#UC-FilterSort-search-results-facets

requires #324

### PR DESCRIPTION
**Pull request by topless**
- formatter and filter naming
- display values to verify sorting
- closes #262

### CODE REVIEW

File: invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/formatters.js [hunk 1]
```diff
@@ -30,11 +30,16 @@ function formatDocumentToTableView(document, seriesPid = null) {
     Updated: toShortDate(fromISO(document.updated)),
   };
   if (!isEmpty(document.metadata)) {
-    serialized['Title'] = document.metadata.title;
-    serialized['Authors'] = document.metadata.authors;
+    assign(serialized, {
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/formatters.js]: Why the use of `assign`?

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentsSearch/DocumentsSearch.js [hunk 2]
```diff
@@ -119,7 +119,7 @@ export class DocumentsSearch extends Component {
         />
         <SortOrder
           values={config.SORT_ORDER}
-          defaultValue={config.SORT_ORDER[0]['value']}
+          defaultValue={config.SORT_ORDER['default_order']}
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentsSearch/DocumentsSearch.js]: you might want to use the `default_order` in all of the `SortOrder` components, in all search pages.

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentsSearch/components/ResultsList/ResultsList.js [hunk 3]
```diff
@@ -22,7 +22,8 @@ export class ResultsList extends Component {
         'Title',
         'Authors',
         'Series',
-        'Items',
+        'Available Items',
+        'Past Loans',
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentsSearch/components/ResultsList/ResultsList.js]: Just IMHO: `Past Loans` -> `N. Loans` ?

File: invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/DocumentsSearch/DocumentsSearch.js [hunk 4]
```diff
@@ -120,6 +120,7 @@ export class DocumentsSearch extends Component {
     return <div>{totalResults} results</div>;
   };
 
+  // FIXME: this overrides whatever configuration we are passing from invenio.
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/DocumentsSearch/DocumentsSearch.js]: do we need to add a FIXME? Can't we fix it?

File: invenio_app_ils/views.py [hunk 5]
```diff
@@ -57,7 +63,7 @@ def _get_documents_ui_config():
     documents_aggs = current_app.config.get('RECORDS_REST_FACETS', {}).get(
         documents_index, {}).get('aggs', {})
     ui_config['documents']['search']['aggs'] = list(documents_aggs.keys())
-
+    import ipdb; ipdb.set_trace()
```
Comment (ntarocco) [inline invenio_app_ils/views.py]: oups!

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentsSearch/components/ResultsList/ResultsList.js (hunk 3 above)
Comment (topless) [reply invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentsSearch/components/ResultsList/ResultsList.js]: I don't mind, but I kept it like this in order to align with all the other variable names. I renamed it to # Loans, N. could be misleading.

**Review by ntarocco — APPROVED** [review]

File: invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/formatters.js [hunk 6]
```diff
@@ -34,7 +34,10 @@ function formatDocumentToTableView(document, seriesPid = null) {
     serialized['Authors'] = document.metadata.authors;
     if (document.metadata.circulation) {
       serialized['Requests'] = document.metadata.circulation.pending_loans;
-      serialized['Items'] = document.metadata.circulation.has_items_for_loan;
+      serialized['Available Items'] =
+        document.metadata.circulation.has_items_for_loan;
+      serialized['# Loans'] =
```
Comment (kpsherva) [inline invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/formatters.js:39]: what does `#` mean?

**Review by kpsherva — APPROVED** [review]

File: invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/formatters.js (hunk 6 above)
Comment (topless) [reply invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/formatters.js:39]: Its a common way to indicate its a number

**Comment by ntarocco** [comment]
If for Karolina was not clear, I think it is less misleading to use any other notation, like N. Or No. Or something else....

**Review by zzacharo — APPROVED** [review]

**Pull request merged**